pkg/apis/flowcontrol/validation/validation_test.go

```diff
@@ -860,6 +876,115 @@ func TestFlowSchemaValidation(t *testing.T) {
+						{
+							Subjects: []flowcontrol.Subject{
+								{
+									Kind: flowcontrol.SubjectKindUser,
+									User: &flowcontrol.UserSubject{Name: "noxu"},
+								},
+							},
+							ResourceRules: []flowcontrol.ResourcePolicyRule{
+								{
+									Verbs:         []string{flowcontrol.VerbAll},
+									APIGroups:     []string{flowcontrol.APIGroupAll},
+									Resources:     []string{flowcontrol.ResourceAll},
+									ClusterScope:  true,
+									ResourceNames: []string{},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: field.ErrorList{
+				field.Required(field.NewPath("spec").Child("rules").Index(0).Child("resourceRules").Index(0).Child("resourceNames"), "resource names must supply at least one resource name"),
+			},
+		},
+		{
+			name: "ResourceNameEvery may not be combined with particulars",
+			flowSchema: &flowcontrol.FlowSchema{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "system-foo",
+				},
+				Spec: flowcontrol.FlowSchemaSpec{
+					MatchingPrecedence: 50,
+					PriorityLevelConfiguration: flowcontrol.PriorityLevelConfigurationReference{
+						Name: "system-bar",
+					},
+					Rules: []flowcontrol.PolicyRulesWithSubjects{
+						{
+							Subjects: []flowcontrol.Subject{
+								{
+									Kind: flowcontrol.SubjectKindUser,
+									User: &flowcontrol.UserSubject{Name: "noxu"},
+								},
+							},
+							ResourceRules: []flowcontrol.ResourcePolicyRule{
+								{
+									Verbs:         []string{flowcontrol.VerbAll},
+									APIGroups:     []string{flowcontrol.APIGroupAll},
+									Resources:     []string{flowcontrol.ResourceAll},
+									ClusterScope:  true,
+									ResourceNames: []string{flowcontrol.ResourceNameEvery, "foo"},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("rules").Index(0).Child("resourceRules").Index(0).Child("resourceNames"), []string{flowcontrol.ResourceNameEvery, "foo"}, "if '*' is present, must not specify other resource names"),
+			},
+		},
+		{
+			name: "ResourceRule.ResourceNames must be well formed",
+			flowSchema: &flowcontrol.FlowSchema{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "system-foo",
+				},
+				Spec: flowcontrol.FlowSchemaSpec{
+					MatchingPrecedence: 50,
+					PriorityLevelConfiguration: flowcontrol.PriorityLevelConfigurationReference{
+						Name: "system-bar",
+					},
+					Rules: []flowcontrol.PolicyRulesWithSubjects{
+						{
+							Subjects: []flowcontrol.Subject{
+								{
+									Kind: flowcontrol.SubjectKindUser,
+									User: &flowcontrol.UserSubject{Name: "noxu"},
+								},
+							},
+							ResourceRules: []flowcontrol.ResourcePolicyRule{
+								{
+									Verbs:         []string{flowcontrol.VerbAll},
+									APIGroups:     []string{flowcontrol.APIGroupAll},
+									Resources:     []string{flowcontrol.ResourceAll},
+									ClusterScope:  true,
+									ResourceNames: []string{"", ""},
```

Comment:
```suggestion
									ResourceNames: []string{"foo", ""},
```

to test the case where one element is valid

Reply:
will address this in https://github.com/kubernetes/kubernetes/pull/94244